src/main/scala/com/google/cloud/spark/bigquery/BigQueryWriteHelper.scala

```diff
@@ -0,0 +1,101 @@
+package com.google.cloud.spark.bigquery
+
+import java.util.UUID
+
+import collection.JavaConverters._
+import com.google.cloud.bigquery.{BigQuery, BigQueryException, FormatOptions, JobInfo, LoadJobConfiguration}
+import com.google.cloud.http.BaseHttpServiceException
+import com.typesafe.scalalogging.StrictLogging
+import org.apache.hadoop.fs.{Path, RemoteIterator}
+import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode}
+
+case class BigQueryWriteHelper(bigQuery:BigQuery,
+                               sqlContext: SQLContext,
+                               saveMode: SaveMode,
+                               options: SparkBigQueryOptions,
+                               data: DataFrame) extends StrictLogging {
+
+  import org.apache.spark.sql.SaveMode
+
+  val conf = sqlContext.sparkSession.sparkContext.hadoopConfiguration
+
+  val temporaryGcsPath = {
+
+    val gcsPathOption = options.temporaryGcsBucket.map(bucket =>
+      s"gs://$bucket/.spark-bigquery-${System.currentTimeMillis()}-${UUID.randomUUID()}")
+    require(gcsPathOption.isDefined, "Temporary GCS path has not been set")
+    val gcsPath = new Path(gcsPathOption.get)
+    val fs = gcsPath.getFileSystem(conf)
+    if (fs.exists(gcsPath)) {
+      throw new IllegalStateException(s"Conflict occurred creating export directory. Path %gcsPath already exists")
+    }
+    gcsPath
+  }
+
+  def writeDataFrameToBigQuery : Unit = {
+    try {
+      val format = options.intermediateFormat.getType.toLowerCase
```

Comment:
it's hacky but I would add cleanTemporaryGcsPath as a shutdownhook in the JVM (SparkContext has a method for it) so that if the user kills their application the data is in theory cleaned up.

Reply:
Added. Notice that the ShutdownHookManager is a Spark private API (See https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/util/ShutdownHookManager.scala), so I've removed this condition from the scalastyle-config file. Changed in https://github.com/GoogleCloudPlatform/spark-bigquery-connector/pull/66/commits/92d7648d9edfb859a70dc4048453fd3fdae73160